src/sentinel.c

```diff
@@ -3319,54 +3320,49 @@ void sentinelConfigSetCommand(client *c) {
-        matches++;
-    }
-
-    if (stringmatch(pattern, "announce-ip", 1)) {
-        addReplyBulkCString(c,"announce-ip");
-        addReplyBulkCString(c,sentinel.announce_ip ? sentinel.announce_ip : "");
-        matches++;
-    }
-
-    if (stringmatch(pattern, "announce-port", 1)) {
-        addReplyBulkCString(c, "announce-port");
-        addReplyBulkLongLong(c, sentinel.announce_port);
-        matches++;
-    }
-
-    if (stringmatch(pattern, "sentinel-user", 1)) {
-        addReplyBulkCString(c, "sentinel-user");
-        addReplyBulkCString(c, sentinel.sentinel_auth_user ? sentinel.sentinel_auth_user : "");
-        matches++;
-    }
-
-    if (stringmatch(pattern, "sentinel-pass", 1)) {
-        addReplyBulkCString(c, "sentinel-pass");
-        addReplyBulkCString(c, sentinel.sentinel_auth_pass ? sentinel.sentinel_auth_pass : "");
-        matches++;
-    }
-
-    if (stringmatch(pattern, "loglevel", 1)) {
-        addReplyBulkCString(c, "loglevel");
-        addReplyBulkCString(c, getLogLevel());
-        matches++;
+    /* Create a dictionary to store the input configs,to avoid adding duplicate twice */
+    dict *d = dictCreate(&externalStringType);
+    for (int i = 3; i < c->argc; i++) {
+        pattern = c->argv[i]->ptr;
+	//we dont want to print duplicates twice
+        if (dictFind(d, pattern)) continue;
+        if (stringmatch(pattern,"resolve-hostnames",1)) {
+            addReplyBulkCString(c,"resolve-hostnames");
+            addReplyBulkCString(c,sentinel.resolve_hostnames ? "yes" : "no");
+            matches++;
+        } else if (stringmatch(pattern, "announce-hostnames", 1)) {
+            addReplyBulkCString(c,"announce-hostnames");
+            addReplyBulkCString(c,sentinel.announce_hostnames ? "yes" : "no");
+            matches++;
+        } else if (stringmatch(pattern, "announce-ip", 1)) {
+            addReplyBulkCString(c,"announce-ip");
+            addReplyBulkCString(c,sentinel.announce_ip ? sentinel.announce_ip : "");
+            matches++;
+        } else if (stringmatch(pattern, "announce-port", 1)) {
+            addReplyBulkCString(c, "announce-port");
+            addReplyBulkLongLong(c, sentinel.announce_port);
+            matches++;
+        } else if (stringmatch(pattern, "sentinel-user", 1)) {
+            addReplyBulkCString(c, "sentinel-user");
+            addReplyBulkCString(c, sentinel.sentinel_auth_user ? sentinel.sentinel_auth_user : "");
+            matches++;
+        } else if (stringmatch(pattern, "sentinel-pass", 1)) {
+            addReplyBulkCString(c, "sentinel-pass");
+            addReplyBulkCString(c, sentinel.sentinel_auth_pass ? sentinel.sentinel_auth_pass : "");
+            matches++;
+        } else if (stringmatch(pattern, "loglevel", 1)) {
+            addReplyBulkCString(c, "loglevel");
+            addReplyBulkCString(c, getLogLevel());
+            matches++;
+        }
+        dictAdd(d, pattern, NULL);
```

Comment:
what if someone passed two different patterns that match the same config twice?
e.g. `sentinel config get log* *level`

Reply:
hi @oranagra , I have missed that scenario earlier, I have took care of it in the latest commit.
thanks.